ui/app/components/copy-button.js

```diff
@@ -1,20 +1,24 @@
-import Component from '@ember/component';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
 import { task, timeout } from 'ember-concurrency';
-import { classNames, classNameBindings } from '@ember-decorators/component';
-import classic from 'ember-classic-decorator';
 
-@classic
-@classNames('copy-button')
-@classNameBindings('inset')
 export default class CopyButton extends Component {
-  clipboardText = null;
-  state = null;
+  @tracked state = null;
+
+  get text() {
+    if (typeof this.args.clipboardText === 'function')
+      return this.args.clipboardText;
+    if (typeof this.args.clipboardText === 'string')
+      return this.args.clipboardText;
+
+    return String(this.args.clipboardText);
+  }
 
   @(task(function* () {
```

Comment:
so it would become 
```
indicateSuccess = task({ restartable: true }, async (t) => { 
  await timeout(2000);
  this.state = null;
});
```
this above the most up to date syntax but if you use version before `v2.3+` you can use decorators like so:
```
@task({ restartable: true })
*indicateSuccess() {
  yield timeout(2000);
  this.state = null;
}
```